### PR TITLE
Sam tov fix docs

### LINKED ISSUE #254: MDSuite modules docs are not there for the calculators.
**Describe the bug**
The MDSuite calculator docs are not in the live documentation.

**To Reproduce**
Look at the calculator docs on readthedocs.

### PR DESCRIPTION
**Pull request by SamTov**
Fix #254 

Attempt at fixing issue #254

#### Summary of additions and changes

* Added documentation for the databases
* Clear issues with incorrect headers and new-lines
* Added proper code-blocks for some code examples
* Fixed some doc strings in the code

#### How to test this pull request

```sh
cd MDSuite/docs
make html
```

#### Comments

We can either change the mdsuite readthedocs branch to check it live or just merge it and see. It builds correctly for me now and all the calculators are there.

### CODE REVIEW

File: mdsuite/database/properties_database.py [hunk 1]
```diff
@@ -232,9 +232,9 @@ def load_data(self, parameters: dict) -> list:
         ----------
         parameters : dict
                 Parameters to be used in the addition, i.e.
-                {"Analysis": "Green_Kubo_Self_Diffusion",
-                 "Subject": "Na",
-                 "data_range": 500}
+                .. code-block::
+
+                   {"Analysis": "Green_Kubo_Self_Diffusion", "Subject": "Na", "data_range": 500}
```
Comment (PythonFZ) [inline mdsuite/database/properties_database.py:237]: the code block looks nice!

File: mdsuite/transformations/map_molecules.py [hunk 2]
```diff
@@ -43,8 +43,11 @@ def __init__(self, experiment: object, molecules: dict):
...
                 Experiment object to work within.
         molecules : dict
-                Molecule dictionary to use as reference. e.g. {'emim': {'smiles': 'CCN1C=C[N+](+C1)C', 'amount': 20},
-                                                                'PF6': {'smiles': 'F[P-](F)(F)(F)(F)F', 'amount': 20}}
+                Molecule dictionary to use as reference. e.g.
+                .. code-block:: python
+
+                   {'emim': {'smiles': 'CCN1C=C[N+](+C1)C', 'amount': 20}, 'PF6': {'smiles': 'F[P-](F)(F)(F)(F)F', 'amount': 20}}
+
                 would be the input for the emim-PF6 ionic liquid.
```
Comment (PythonFZ) [inline mdsuite/transformations/map_molecules.py:51]: I just build the docs and `MDSuite/docs/build/html/_transformations/map_molecules.html` is just an empty page

**Review by PythonFZ — CHANGES_REQUESTED** [review]
Some code blocks aren't working.

In addition I get these warnings when building the docs:

```
WARNING: error while formatting signature for mdsuite.calculators.einstein_diffusion_coefficients.EinsteinDiffusionCoefficients: Handler <function mangle_signature at 0x000001E5891C7430> for event 'autodoc-process-signature' threw a
n exception (exception: mdsuite.calculators.calculator.Calculator class is not a item name in 'Class for the Einstein diffusion coefficient implementation\n\nDescription: This module contains the code for the Einstein diffusion coef
ficient class.\nThis class is called by the Experiment class and instantiated when the user calls the\nExperiment.einstein_diffusion_coefficients method. The methods in class can then be\ncalled by the Experiment.einstein_diffusion_
coefficients method and all necessary\ncalculations performed.\n\nAttributes\n----------\nexperiment :  object\n        Experiment class to call from\nspecies : list\n        Which species to perform the analysis on\nx_label : str\n
        X label of the tensor_values when plotted\ny_label : str\n        Y label of the tensor_values when plotted\nanalysis_name : str\n        Name of the analysis\nloaded_property : str\n        Property loaded from the database
_path for the analysis\n\nSee Also\n--------\nmdsuite.calculators.calculator.Calculator class\n\nExamples\n--------\nexperiment.run_computation.EinsteinDiffusionCoefficients(data_range=500, plot=True, correlation_time=10)')
WARNING: error while formatting signature for mdsuite.calculators.einstein_helfand_ionic_conductivity.EinsteinHelfandIonicConductivity: Handler <function mangle_signature at 0x000001E5891C7430> for event 'autodoc-process-signature'
threw an exception (exception: mdsuite.calculators.calculator.Calculator class is not a item name in 'Class for the Einstein-Helfand Ionic Conductivity\n\nAttributes\n----------\nexperiment :  object\n        Experiment class to cal
l from\nx_label : str\n        X label of the tensor_values when plotted\ny_label : str\n        Y label of the tensor_values when plotted\nanalysis_name : str\n        Name of the analysis\nloaded_property : str\n        Property l
oaded from the database_path for the analysis\n\nSee Also\n--------\nmdsuite.calculators.calculator.Calculator class\n\nExamples\n--------\nexperiment.run_computation.EinsteinHelfandTIonicConductivity(data_range=500, plot=True, corr
elation_time=10)')
WARNING: error while formatting signature for mdsuite.calculators.einstein_helfand_thermal_conductivity.EinsteinHelfandThermalConductivity: Handler <function mangle_signature at 0x000001E5891C7430> for event 'autodoc-process-signatu
re' threw an exception (exception: mdsuite.calculators.calculator.Calculator class is not a item name in 'Class for the Einstein-Helfand Ionic Conductivity\n\nAttributes\n----------\nexperiment :  object\n        Experiment class to
 call from\nx_label : str\n        X label of the tensor_values when plotted\ny_label : str\n        Y label of the tensor_values when plotted\nanalysis_name : str\n        Name of the analysis\nloaded_property : str\n        Proper
ty loaded from the database_path for the analysis\n\nSee Also\n--------\nmdsuite.calculators.calculator.Calculator class\n\nExamples\n--------\nexperiment.run_computation.EinsteinHelfandTThermalConductivity(data_range=500, plot=True
, correlation_time=10)')
WARNING: error while formatting signature for mdsuite.calculators.green_kubo_self_diffusion_coefficients.GreenKuboSelfDiffusionCoefficients: Handler <function mangle_signature at 0x000001E5891C7430> for event 'autodoc-process-signat
ure' threw an exception (exception: mdsuite.calculators.calculator.Calculator class is not a item name in 'Class for the Green-Kubo diffusion coefficient implementation\nAttributes\n----------\nexperiment :  object\n        Experime
nt class to call from\nspecies : list\n        Which species to perform the analysis on\nx_label : str\n        X label of the tensor_values when plotted\ny_label : str\n        Y label of the tensor_values when plotted\nanalysis_na
me : str\n        Name of the analysis\nloaded_property : str\n        Property loaded from the database_path for the analysis\n\nSee Also\n--------\nmdsuite.calculators.calculator.Calculator class\n\nExamples\n--------\nexperiment.
run_computation.GreenKuboSelfDiffusionCoefficients(data_range=500, plot=True, correlation_time=10)')
WARNING: error while formatting signature for mdsuite.calculators.green_kubo_ionic_conductivity.GreenKuboIonicConductivity: Handler <function mangle_signature at 0x000001E5891C7430> for event 'autodoc-process-signature' threw an exc
eption (exception: mdsuite.calculators.calculator.Calculator class is not a item name in 'Class for the Green-Kubo ionic conductivity implementation\n\nAttributes\n----------\nexperiment :  object\n        Experiment class to call f
rom\nx_label : str\n        X label of the tensor_values when plotted\ny_label : str\n        Y label of the tensor_values when plotted\nanalysis_name : str\n        Name of the analysis\nloaded_property : str\n        Property load
ed from the database_path for the analysis\n\nSee Also\n--------\nmdsuite.calculators.calculator.Calculator class\n\nExamples\n--------\nexperiment.run_computation.GreenKuboIonicConductivity(data_range=500, plot=True, correlation_ti
me=10)')
WARNING: error while formatting signature for mdsuite.calculators.green_kubo_thermal_conductivity.GreenKuboThermalConductivity: Handler <function mangle_signature at 0x000001E5891C7430> for event 'autodoc-process-signature' threw an
 exception (exception: mdsuite.calculators.calculator.Calculator class is not a item name in 'Class for the Green-Kubo Thermal conductivity implementation\n\nAttributes\n----------\nexperiment :  object\n        Experiment class to
call from\nx_label : str\n        X label of the tensor_values when plotted\ny_label : str\n        Y label of the tensor_values when plotted\nanalysis_name : str\n        Name of the analysis\n\nSee Also\n--------\nmdsuite.calculat
ors.calculator.Calculator class\n\nExamples\n--------\nexperiment.run_computation.GreenKuboThermalConductivity(data_range=500, plot=True, correlation_time=10)')
WARNING: error while formatting signature for mdsuite.calculators.kirkwood_buff_integrals.KirkwoodBuffIntegral: Handler <function mangle_signature at 0x000001E5891C7430> for event 'autodoc-process-signature' threw an exception (exce
ption: mdsuite.calculators.calculator.Calculator class is not a item name in 'Class for the calculation of the Kikrwood-Buff integrals\n\nAttributes\n----------\nexperiment : class object\n                    Class object of the exp
eriment.\ndata_range : int (default=500)\n                    Range over which the property should be evaluated. This is not applicable to the current\n                    analysis as the full rdf will be calculated.\nx_label : str\
n                    How to label the x axis of the saved plot.\ny_label : str\n                    How to label the y axis of the saved plot.\nanalysis_name : str\n                    Name of the analysis. used in saving of the ten
sor_values and figure.\nfile_to_study : str\n                    The tensor_values file corresponding to the rdf being studied.\ndata_files : list\n                    list of files to be analyzed.\nrdf = None : list\n
      rdf tensor_values being studied.\nradii = None : list\n                    radii tensor_values corresponding to the rdf.\nspecies_tuple : list\n                    A list of species combinations being studied.\n\nSee Also\n---
-----\nmdsuite.calculators.calculator.Calculator class\n\nExamples\n--------\nexperiment.run_computation.KirkwoodBuffIntegral()')
WARNING: error while formatting signature for mdsuite.calculators.nernst_einstein_ionic_conductivity.NernstEinsteinIonicConductivity: Handler <function mangle_signature at 0x000001E5891C7430> for event 'autodoc-process-signature' th
rew an exception (exception: mdsuite.calculators.calculator.Calculator class is not a item name in 'Class for the calculation of the Nernst-Einstein ionic conductivity\n\nSee Also\n--------\nmdsuite.calculators.calculator.Calculator
 class\n\nExamples\n--------\nexperiment.run_computation.NernstEinsteinIonicConductivity()')
WARNING: error while formatting signature for mdsuite.calculators.potential_of_mean_force.PotentialOfMeanForce: Handler <function mangle_signature at 0x000001E5891C7430> for event 'autodoc-process-signature' threw an exception (exce
ption: mdsuite.calculators.calculator.Calculator class is not a item name in 'Class for the calculation of the potential of mean-force\n\nThe potential of mean-force is a measure of the binding strength between atomic species in a e
xperiment.\n\nAttributes\n----------\nexperiment : class object\n                    Class object of the experiment.\ndata_range : int (default=500)\n                    Range over which the property should be evaluated. This is not
 applicable to the current\n                    analysis as the full rdf will be calculated.\nx_label : str\n                    How to label the x axis of the saved plot.\ny_label : str\n                    How to label the y axis
of the saved plot.\nanalysis_name : str\n                    Name of the analysis. used in saving of the tensor_values and figure.\nfile_to_study : str\n                    The tensor_values file corresponding to the rdf being studi
ed.\ndata_files : list\n                    list of files to be analyzed.\nrdf = None : list\n                    rdf tensor_values being studied.\nradii = None : list\n                    radii tensor_values corresponding to the rd
f.\nspecies_tuple : list\n                    A list of species combinations being studied.\n\nSee Also\n--------\nmdsuite.calculators.calculator.Calculator class\n\nExamples\n--------\nexperiment.run_computation.PotentialOfMeanForc
e(savgol_order = 2, savgol_window_length = 17)')
WARNING: error while formatting signature for mdsuite.calculators.radial_distribution_function.RadialDistributionFunction: Handler <function mangle_signature at 0x000001E5891C7430> for event 'autodoc-process-signature' threw an exce
ption (exception: mdsuite.calculators.calculator.Calculator class is not a item name in 'Class for the calculation of the radial distribution function\n\nAttributes\n----------\nexperiment :  object\n        Experiment class to call
 from\ndata_range :\n        Number of configurations to use in each ensemble\nx_label : str\n        X label of the tensor_values when plotted\ny_label : str\n        Y label of the tensor_values when plotted\nanalysis_name : str\n
        Name of the analysis\nloaded_property : str\n        Property loaded from the database_path for the analysis\nminibatch: int, default None\n        Size of a individual minibatch, if set. By default mini-batching is not appl
ied\n\nSee Also\n--------\nmdsuite.calculators.calculator.Calculator class\n\nExamples\n--------\nexperiment.run_computation.RadialDistributionFunction(number_of_configurations = 500, minibatch = 10, start = 0,\n
                                   stop = 1000, number_of_bins = 100, use_tf_function = False)')
WARNING: error while formatting signature for mdsuite.calculators.structure_factor.StructureFactor: Handler <function mangle_signature at 0x000001E5891C7430> for event 'autodoc-process-signature' threw an exception (exception: mdsui
te.calculators.calculator.Calculator class is not a item name in "Class for the calculation of the total structure factor for X-ray scattering\nusing the Faber-Ziman partial structure factors. This analysis is valid for a magnitude
of the X-ray\nscattering vector Q < 25 * 1/Angstrom\nExplicitly equations 9, 10 and 11 of the paper\n'DFT Accurate Interatomic Potential for Molten NaCl from MachineLearning' from\nSamuel Tovey, Anand Narayanan Krishnamoorthy, Ganes
h Sivaraman, Jicheng Guo,\nChris Benmore,Andreas Heuer, and Christian Holm are implemented.\n\nAttributes\n----------\nexperiment : class object\n                    Class object of the experiment.\nx_label : str\n
  How to label the x axis of the saved plot.\ny_label : str\n                    How to label the y axis of the saved plot.\nanalysis_name : str\n                    Name of the analysis. used in saving of the tensor_values and figu
re.\nfile_to_study : str\n                    The tensor_values file corresponding to the rdf being studied.\ndata_files : list\n                    list of files to be analyzed.\nrdf = None : list\n                    rdf tensor_va
lues being studied.\nradii = None : list\n                    radii tensor_values corresponding to the rdf.\n\nSee Also\n--------\nmdsuite.calculators.calculator.Calculator class\n\nExamples\n--------\nexperiment.run_computation.Str
uctureFactor()\n\nNotes\n-----\nIn order to use the structure factor calculator both the masses and the charges of each species must be present.\nIf they are not correct, the structure factor will not work.")
C:\Users\fabia\Nextcloud\DATA\JupyterProjects\MDSuite\mdsuite\database\properties_database.py:docstring of mdsuite.database.properties_database.PropertiesDatabase.load_data:4: WARNING: Unexpected indentation.
C:\Users\fabia\Nextcloud\DATA\JupyterProjects\MDSuite\mdsuite\database\properties_database.py:docstring of mdsuite.database.properties_database.PropertiesDatabase.load_data:7: WARNING: Block quote ends without a blank line; unexpect
ed unindent.
WARNING: autodoc: failed to import class 'map_molecules.MolecularMap' from module 'mdsuite.transformations'; the following exception was raised:
No module named 'pysmiles'
WARNING: error while formatting signature for mdsuite.calculators.coordination_number_calculation.CoordinationNumbers: Handler <function mangle_signature at 0x000001E5891C7430> for event 'autodoc-process-signature' threw an exceptio
n (exception: mdsuite.calculators.calculator.Calculator class is not a item name in 'Class for the calculation of coordination numbers\n\nAttributes\n----------\nexperiment : class object\n                    Class object of the exp
eriment.\n\ndata_range : int (default=500)\n                    Range over which the property should be evaluated. This is not applicable to the current\n                    analysis as the full rdf will be calculated.\nx_label : st
r\n                    How to label the x axis of the saved plot.\ny_label : str\n                    How to label the y axis of the saved plot.\nanalysis_name : str\n                    Name of the analysis. used in saving of the t
ensor_values and figure.\nfile_to_study : str\n                    The tensor_values file corresponding to the rdf being studied.\nrdf = None : list\n                    rdf tensor_values being studied.\nradii = None : list\n
             radii tensor_values corresponding to the rdf.\nintegral_data : list\n                    integrated rdf tensor_values.\nspecies_tuple : list\n                    A list of species combinations being studied.\nindices :
list\n                    A list of indices which correspond to to correct coordination numbers.\n\nSee Also\n--------\nmdsuite.calculators.calculator.Calculator class\n\nExamples\n--------\nexperiment.run_computation.CoordinationNu
mbers(savgol_order = 2, savgol_window_length = 17)')
C:\Users\fabia\Nextcloud\DATA\JupyterProjects\MDSuite\mdsuite\transformations\unwrap_coordinates.py:docstring of mdsuite.transformations.unwrap_coordinates.CoordinateUnwrapper:1: WARNING: duplicate object description of mdsuite.tran
sformations.unwrap_coordinates.CoordinateUnwrapper, other instance in _transformations/unwrap_coordinates, use :noindex: for one of them
C:\Users\fabia\Nextcloud\DATA\JupyterProjects\MDSuite\mdsuite\transformations\unwrap_coordinates.py:docstring of mdsuite.transformations.unwrap_coordinates.CoordinateUnwrapper:4: WARNING: duplicate object description of mdsuite.tran
sformations.unwrap_coordinates.CoordinateUnwrapper.experiment, other instance in _transformations/unwrap_coordinates, use :noindex: for one of them
C:\Users\fabia\Nextcloud\DATA\JupyterProjects\MDSuite\mdsuite\transformations\unwrap_coordinates.py:docstring of mdsuite.transformations.unwrap_coordinates.CoordinateUnwrapper:10: WARNING: duplicate object description of mdsuite.tra
nsformations.unwrap_coordinates.CoordinateUnwrapper.species, other instance in _transformations/unwrap_coordinates, use :noindex: for one of them
C:\Users\fabia\Nextcloud\DATA\JupyterProjects\MDSuite\mdsuite\transformations\unwrap_coordinates.py:docstring of mdsuite.transformations.unwrap_coordinates.CoordinateUnwrapper:16: WARNING: duplicate object description of mdsuite.tra
nsformations.unwrap_coordinates.CoordinateUnwrapper.center_box, other instance in _transformations/unwrap_coordinates, use :noindex: for one of them
C:\Users\fabia\Nextcloud\DATA\JupyterProjects\MDSuite\mdsuite\transformations\unwrap_coordinates.py:docstring of mdsuite.transformations.unwrap_coordinates.CoordinateUnwrapper:23: WARNING: duplicate object description of mdsuite.tra
nsformations.unwrap_coordinates.CoordinateUnwrapper.scale_function, other instance in _transformations/unwrap_coordinates, use :noindex: for one of them
C:\Users\fabia\Nextcloud\DATA\JupyterProjects\MDSuite\mdsuite\transformations\unwrap_coordinates.py:docstring of mdsuite.transformations.unwrap_coordinates.CoordinateUnwrapper.run_transformation:1: WARNING: duplicate object descript
ion of mdsuite.transformations.unwrap_coordinates.CoordinateUnwrapper.run_transformation, other instance in _transformations/unwrap_coordinates, use :noindex: for one of them
```

File: mdsuite/database/properties_database.py [hunk 3]
```diff
@@ -232,9 +232,9 @@ def load_data(self, parameters: dict) -> list:
         ----------
         parameters : dict
                 Parameters to be used in the addition, i.e.
```
[the remark this answers is not in the document]
Comment (SamTov) [reply mdsuite/database/properties_database.py:234]: Yes after that brief lesson in latin I believe it should be. I admire the attention to detail

File: mdsuite/transformations/map_molecules.py (hunk 2 above)
Comment (SamTov) [reply mdsuite/transformations/map_molecules.py:51]: If you look at the warnings one of them is that it did not import

Comment (SamTov) [reply mdsuite/transformations/map_molecules.py:51]: It compiles for me. What version of sphinx?

File: mdsuite/project.py [hunk 4]
```diff
@@ -282,9 +282,10 @@ def get_properties(self, parameters: dict, experiments: list = None):
         ----------
         parameters : dict
                 Parameters to be used in the addition, i.e.
-                {"Analysis": "Green_Kubo_Self_Diffusion",
-                 "Subject": "Na",
-                 "data_range": 500}
+                .. code-block:: python
+
+                   {"Analysis": "Green_Kubo_Self_Diffusion",  "Subject": "Na", "data_range": 500}
+
```
[the remark this answers is not in the document]
Comment (SamTov) [reply mdsuite/project.py:288]: I believe it needed an additional space

File: mdsuite/transformations/map_molecules.py (hunk 2 above)
Comment (PythonFZ) [reply mdsuite/transformations/map_molecules.py:51]: `Running Sphinx v4.0.1`

Comment (SamTov) [reply mdsuite/transformations/map_molecules.py:51]: hmm me too

Comment (PythonFZ) [reply mdsuite/transformations/map_molecules.py:51]: I downloaded the GitHub build documentation, which is all that matters. The map molecules is build there, but the code block does not seem to work

Comment (PythonFZ) [reply mdsuite/transformations/map_molecules.py:51]: I guess it is also missing the blank space above. If that is added it should be solved

Comment (SamTov) [reply mdsuite/transformations/map_molecules.py]: Please check again.

**Comment by SamTov** [comment]
It seems there will always be errors when compiling the docs. Someone will need to spend more than an hour trying to fix it up to have none, hopefully however the calculators now show up when we merge into main.

File: mdsuite/transformations/map_molecules.py (hunk 2 above)
Comment (PythonFZ) [reply mdsuite/transformations/map_molecules.py]: Still not working

Comment (SamTov) [reply mdsuite/transformations/map_molecules.py]: So for me it does not update at all. I added new lines to the doc string and it has refused to update. I looked into the link and it seems okay.

Comment (PythonFZ) [reply mdsuite/transformations/map_molecules.py]: This is how it looks for me

image

But I assume it should be similar to the e.g. the project class one
image

Comment (PythonFZ) [reply mdsuite/transformations/map_molecules.py]: I fixed it in 955c98c ... I guess we both looked at the wrong place

**Review by PythonFZ — APPROVED** [review]
Sphinx can be a little confusing on where to put the docs for classes. Does it even look at the init?
Either way we can merge this now.

**Pull request merged**